### ISSUE TITLE
Ongoing Minor Issue List SpongeDocs Edition

### ISSUE THREAD

**Issue by Inscrutable** [SpongePowered/SpongeDocs#698]
Rather than push trivial fixes as PRs, we prefer to collect them together and push them in batches.
Please let us know of any minor corrections, nitpicks, and errors in spelling, grammar or punctuation that are needed, and we shall do our best to see they are dealt with.

This is a reincarnation of the humungous list at #437 - thanks to everyone so far!

**Comment by ST-DDT** [comment]
* [ ] https://github.com/SpongePowered/SpongeDocs/blame/stable/source/server/management/exploit-patches.rst#L16

> **Exploits Patched implemented in Sponge**

Should be

> **Exploit Patches Implemented In Sponge**

or 

> **Exploits Patched In Sponge**

**Comment by ST-DDT** [comment]
* [ ] https://github.com/SpongePowered/SpongeDocs/blame/stable/source/plugin/commands/childcommands.rst#L63

> If a command has child commands, a :javadoc:`CommandExecutor`, set through
:javadoc:`CommandSpec.Builder#executor(CommandExecutor)` and their associated
:javadoc:`CommandSpec.Builder#arguments(CommandElement)` are optional. The behavior of error in selection and
argument parsing of child commands is dependent on whether this parent executor exists.

This section is hard to understand

**Comment by ST-DDT** [comment]
* [ ] https://github.com/SpongePowered/SpongeDocs/blame/stable/source/plugin/commands/commandcallable.rst#L63

> `public List<String> getSuggestions(CommandSource source, String arguments) throws CommandException {`

This method is now defined like this:

> `public List<String> getSuggestions(CommandSource source, String arguments, @Nullable  Location<World> targetPosition) throws CommandException {`

**Comment by ST-DDT** [comment]
* [ ] https://github.com/SpongePowered/SpongeDocs/blame/stable/source/plugin/items/creating.rst#L111

> `try (StackFrame frame = Sponge.getCauseStackManager().pushStackFrame()) {`

Should be

> `try (StackFrame frame = Sponge.getCauseStackManager().pushCauseFrame()) {`

**Comment by ST-DDT** [comment]
* [ ] Differently named references to `PluginContainer`s / `Plugin`s

* `plugin`
* `myplugin`
* `yourPluginInstance`

Should probably use `plugin` all the time.

**Comment by Grauldon** [comment]
- [ ] [Testing Your Plugin](https://docs.spongepowered.org/stable/en/plugin/workspace/idea.html#testing-your-plugin)

* Under `Tasks` -> `Build`, click on `jar` 
should be 
* Under `Tasks` -> `Build`, **double-** click on `jar`

**Comment by Grauldon** [comment]
- [x] Update Ore entry in [Sponge Glossary ](https://docs.spongepowered.org/stable/en/about/glossary.html) to just `The Official Sponge plugin hosting site.` (Could provide a link to the site.)

**Comment by Grauldon** [comment]
- [ ] Locate and update branch names from `stable` and `bleeding` to `api-#` as appropriate.

**Comment by MinusKube** [comment]
- [ ] https://github.com/SpongePowered/SpongeDocs/blame/stable/source/plugin/optional/basic.rst#L15
- [ ] https://github.com/SpongePowered/SpongeDocs/blame/stable/source/plugin/optional/basic.rst#L94

These two titles are available for translation on Crowdin, but are not getting translated on the final page, example: https://docs.spongepowered.org/stable/fr/plugin/optional/basic.html#implicit-nullable-contracts-and-why-they-suck

**Comment by Inscrutable** [comment]
Good catch. That's another issue caused by a title leading with a number and fooling the ReST / Sphinx formatting. We edited out the last few, and this one got missed somehow.

**Comment by ST-DDT** [comment]
* [ ] https://github.com/SpongePowered/SpongeDocs/blame/stable/source/server/spongineer/logs.rst#L8

> ![screenshot](https://media.discordapp.net/attachments/206524927904317450/588884733283401730/unknown.png)
> 
> It looks like it is meant to say "Table of Contents"

Originally reported by 3TUSK on Discord

**Comment by mjra007** [comment]
This was pointed out originally on [discord](https://discordapp.com/channels/142425412096491520/142425521391665153/755406301730701396). There is a very minor "issue" with a code block to [access a data manipulator for a block](https://github.com/SpongePowered/SpongeDocs/blob/stable/source/plugin/blocks/accessing.rst#block-data-manipulators).  


    import org.spongepowered.api.data.key.Keys;
    import org.spongepowered.api.data.manipulator.mutable.block.DirectionalData;

    public boolean isFacingNorth(Location<World> blockLoc) {
        Optional<DirectionalData> optionalData = blockLoc.get(DirectionalData.class);
        if (!optionalData.isPresent()) {
            return false;
        }
        DirectionalData data = optionalData.get();
        if (data.get(Keys.DIRECTION).get().equals(Direction.NORTH)) {
            return true;
        }
        return false;
    }

It might be that the code gets optimized after it is compiled, especially the last if statement (don't know). However, this could be easily be written like this:

    import org.spongepowered.api.data.key.Keys;
    import org.spongepowered.api.data.manipulator.mutable.block.DirectionalData;

    public boolean isFacingNorth(Location<World> blockLoc) {
        return blockLoc.get(DirectionalData.class)
                .filter(dir -> dir.direction().get().equals(Direction.NORTH))
                .isPresent();
    }
